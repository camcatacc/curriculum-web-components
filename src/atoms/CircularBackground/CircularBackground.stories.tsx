// Modules
import React from "react";

// Storybook Modules
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Elements
import CircularBackground from "atoms/CircularBackground/CircularBackground";

// Styles
import { secondary } from "styles/customColors";

// Template
const Template: ComponentStory<typeof CircularBackground> = (args) => (
	<div className="flex">
		<CircularBackground {...args} />
	</div>
);

// Variants
export const Standard = Template.bind({});
Standard.args = {
	children: <p style={{ color: "white", margin: "0.5rem" }}>This is a large circle</p>,
	backgroundColor: secondary
};

// Export
export default {
	title: "Atoms/CircularBackground",
	component: CircularBackground
} as ComponentMeta<typeof CircularBackground>;
