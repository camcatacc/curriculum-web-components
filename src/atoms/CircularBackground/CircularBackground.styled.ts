// Modules
import styled from "styled-components";

// Definitions
import type { CircularBackgroundProps } from "atoms/CircularBackground/CircularBackground";

// Components
export const Container = styled.div<Omit<CircularBackgroundProps, "children">>`
	background-color: ${({ backgroundColor }) => backgroundColor};
	height: ${({ height }) => height};
	width: ${({ width }) => width};
	padding: ${({ padding }) => padding};
`;
