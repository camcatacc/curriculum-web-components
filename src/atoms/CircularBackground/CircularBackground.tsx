// Modules
import React from "react";

// Styles
import * as Styled from "atoms/CircularBackground/CircularBackground.styled";

// Definitions
export interface CircularBackgroundProps {
	children: React.ReactNode;
	backgroundColor?: string;
	height?: string | number;
	width?: string | number;
	padding?: string | number;
}

// Element
const CircularBackground = ({ children, backgroundColor, height, width, padding }: CircularBackgroundProps) => {
	return (
		<Styled.Container data-testid="container" className="rounded-full relative flex" backgroundColor={backgroundColor} height={height} width={width} padding={padding}>
			<div className="h-full flex justify-center items-center">{children}</div>
			{/** This is a dummy so that the height is the same than the width */}
			<div style={{ marginTop: "100%" }} />
		</Styled.Container>
	);
};

export default CircularBackground;
