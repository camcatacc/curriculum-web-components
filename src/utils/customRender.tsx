import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
