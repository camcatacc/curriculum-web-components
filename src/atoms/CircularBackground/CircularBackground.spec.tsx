// Modules
import React from "react";
import { fireEvent } from "@testing-library/react";
import { customRender } from "utils/customRender";
import "@testing-library/jest-dom/extend-expect";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import CircularBackground from "atoms/CircularBackground/CircularBackground";

// Definitions
import type { CircularBackgroundProps } from "atoms/CircularBackground/CircularBackground";

const defaultProps: CircularBackgroundProps = {
	children: <p>Circular Background</p>,
	backgroundColor: "black"
};
// Rendering
const renderComponent = (props = {}) => {
	const combinedProps = { ...defaultProps, ...props };
	return customRender(<CircularBackground {...combinedProps} />);
};

describe("CircularBackground", () => {
	it("Snapshot", async () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});

	it("Renders the required props as stated", () => {
		const { getByText, getByTestId } = renderComponent();
		expect(getByTestId("container")).toHaveStyle("background-color: black");
		expect(getByText("Circular Background")).toBeInTheDocument();
	});
});
// Tests
expect.extend(toHaveNoViolations);
