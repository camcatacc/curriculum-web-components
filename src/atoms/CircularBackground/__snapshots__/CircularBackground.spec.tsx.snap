// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularBackground Snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL fQxqua rounded-full relative flex"
    data-testid="container"
  >
    <div
      class="h-full flex justify-center items-center"
    >
      <p>
        Circular Background
      </p>
    </div>
    <div
      style="margin-top: 100%;"
    />
  </div>
</div>
`;
